spec/unit/pops/lookup/lookup_spec.rb

```diff
@@ -28,6 +28,7 @@ module Lookup
           a: a (from global)
           d: d (from global)
           mod::e: mod::e (from global)
+          other::y: should be removed
```

Comment:
Maybe add a key to a module's `hiera.yaml`, where the key doesn't match the module's name. Looking at
https://github.com/puppetlabs/puppet/blob/fa213f5808c271c9e30fce04779489806fc93da7/spec/unit/pops/lookup/lookup_spec.rb#L57-L58

You could add an entry like the following:

```ruby
        'common.yaml' => <<-YAML.unindent
          ....
          other::c: other::c (from module)
        YAML
```

And then add tests around here https://github.com/puppetlabs/puppet/blob/fa213f5808c271c9e30fce04779489806fc93da7/spec/unit/pops/lookup/lookup_spec.rb#L192-L194

to test that if you do `Lookup.lookup('other::c', ...)` then it should return nil. And if you lookup an unqualified `Lookup.lookup('c', ...)` then it should return the `mod::c (from module)`, since the value from `other::c` should be ignored.

Prior to your changes, I would expect `other::c (from) module` to be returned in both cases, since it wasn't filtered out.

Reply:
It doesn't seem like unqualified lookups work in tests, both in my PR or in `main`. I'll dig into it a bit more and see if that's expected, it's a bug, or if there's something wrong with the tests.